1-exercises/B-while-loop/exercise.js

```diff
@@ -7,8 +7,19 @@
 
 function evenNumbers(n) {
     // TODO
+    let i = 0, str = ``;
+    while (n > 0) {
+        (n === 1) ? str += `${i}` :
+        str += `${i},`;
+        i += 2;
+        n--;
+    }
+    console.log(str);
```

Comment:
Nice, this will work - can you also think about a way of returning the even numbers sequence by counting from 0 to n?

Reply:
I like this alternative approach!

In general, though, I would encourage you to avoid string concatenation when possible, as in javascript and other languages, strings are immutable, and concatenating them is an `O(n^2)` operation, where `n` is the combined size of the two strings, therefore quite expensive. They will also need to get disposed by the garbage collector when you stop referencing them, making your code quite slow.

A more efficient approach is to use array methods instead, like in [this example](https://github.com/CodeYourFuture/JavaScript-Core-1-Coursework-Week3-London8/blob/d1563d69fc66212c27517da938ad676297786c4f/1-exercises/B-while-loop/exercise.js).

Reply:
Thanks for the elaboration @gixo!!! It's good to know that concatenation has a quadratic complexity. I'll try to keep that in mind for future assignments and fix it here too if I have extra time!